.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,64 @@
+name: Deploy to GitHub Pages
```

Comment:
GitHub themselves provide starter workflows for publishing to GitHub Pages using various static site generators (unfortunately not including Pelican), the way I've designed this workflow is similar to these: https://github.com/actions/starter-workflows/tree/main/pages

Reply:
I'm not sure how I feel about hosting a workflow file under `.github/workflows` that is *not* used in this repository, just to make it reusable. I'm just saying this before I start reviewing, which will have to wait for some free time for me.

Reply:
Reusable workflows have to be under `.github/workflows/` otherwise GitHub won't recognise them. Unfortunately they can't be in a subdirectory of `.github/workflows/` either ([docs](https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow)).

It could be moved into a `.github/workflows/` directory of a separate `getpelican/reusable-workflows` repo, but the nice thing about having the workflow be in the `getpelican/pelican` repo itself is that you can change the workflow and its associated documentation at the same time, in one PR.

Reply:
> I'm not sure how I feel about hosting a workflow file under `.github/workflows` that is _not_ used in this repository, just to make it reusable. I'm just saying this before I start reviewing, which will have to wait for some free time for me.

That is a valid point. On the other side, I think it would be good to have (even if I don't use it). Many projects on GH just use a different approach: They create a dedicated repo for that action in same org.
One could do the same and have the minimal documentation as a pelican site deployed as github page from that said repo.
Benefits:
  - this repo is not cluttered with unused code
  - the action is tested and used with it's own documentation
 
Con: 
  - another repo to maintain